src/lib/marshal/workflow/writer.ts

```diff
@@ -183,7 +184,11 @@ const compileExtractionSettings = (
           map.set([...parts, key], extractableFields[key]);
         }
 
-        compileRecursively(val, [...parts, key]);
+        // Recursively exam current field for any additionally extractable data
+        // within, except for disallowed fields
+        if (!NON_RECURSIVELY_TRAVERSABLE_FIELDS_FOR_EXTRACTION.has(key)) {
+          compileRecursively(val, [...parts, key]);
+        }
```

Comment:
Without this check, we'll recurse down into an if-else step's `branches` list and prematurely create a whole mess of extracted content for nested steps.